fili-core/src/main/java/com/yahoo/bard/webservice/data/metric/protocol/ProtocolMetricImpl.java

```diff
@@ -75,6 +75,19 @@ public ProtocolSupport getProtocolSupport() {
         return protocolSupport;
     }
 
+    @Override
+    public ProtocolMetric withLogicalMetricInfo(LogicalMetricInfo info) {
+        TemplateDruidQuery newTdq = getTemplateDruidQuery();
+        if (newTdq != null) {
```

Comment:
I wonder if maybe we should toss this into a helper method on LogicalMetricImpl. Pretty much
every child class is going to have to do this, so having a null-safe `renameTemplateDruidQuery(LogicalMetricInfo info)`
may be worthwhile. Then, people can just do `return new MaMetric(info, renameTemplateDruidQuery(info)`, maOtherStuff)`.

Reply:
👍 This seems like a good idea.